featuretools/tests/entityset_tests/test_ww_es.py

```diff
@@ -206,3 +211,284 @@ def test_add_last_time_indexes(es):
+        # Dask and Koalas do not error on invalid type conversion until compute
+        error_msg = 'Error converting datatype for age from type object to type int64. Please confirm the underlying data is consistent with logical type Integer.'
+        with pytest.raises(ww.exceptions.TypeConversionError, match=error_msg):
+            es.update_dataframe(dataframe_id='customers', df=incompatible_dtype_df)
+
+
+@pytest.fixture()
+def latlong_df_pandas():
+    latlong_df = pd.DataFrame({
+        'string_tuple': pd.Series(['(1, 2)', '(3, 4)']),
+        'bracketless_string_tuple': pd.Series(['1, 2', '3, 4']),
+        'list_strings': pd.Series([['1', '2'], ['3', '4']]),
+        'combo_tuple_types': pd.Series(['[1, 2]', '(3, 4)']),
+    })
+    latlong_df.set_index('string_tuple', drop=False, inplace=True)
+    latlong_df.index.name = None
+    return latlong_df
+
+
+@pytest.fixture()
+def latlong_df_dask(latlong_df_pandas):
+    dd = pytest.importorskip('dask.dataframe', reason='Dask not installed, skipping')
+    return dd.from_pandas(latlong_df_pandas, npartitions=2)
+
+
+@pytest.fixture()
+def latlong_df_koalas(latlong_df_pandas):
+    ks = pytest.importorskip('databricks.koalas', reason='Koalas not installed, skipping')
+    return ks.from_pandas(latlong_df_pandas.applymap(lambda tup: list(tup) if isinstance(tup, tuple) else tup))
+
+
+@pytest.fixture(params=['latlong_df_pandas', 'latlong_df_dask', 'latlong_df_koalas'])
+def latlong_df(request):
+    return request.getfixturevalue(request.param)
+
+
+def test_update_dataframe_data_transformation(latlong_df):
+    initial_df = latlong_df.copy()
+    initial_df.ww.init(index='string_tuple', logical_types={col_name: 'LatLong' for col_name in initial_df.columns})
+    es = EntitySet()
+    es.add_dataframe(dataframe_id='latlongs', dataframe=initial_df)
+
+    df = to_pandas(es['latlongs'])
+    expected_val = (1, 2)
+    if ks and isinstance(es['latlongs'], ks.DataFrame):
+        expected_val = [1, 2]
+    for col in latlong_df.columns:
+        series = df[col]
+        assert series.iloc[0] == expected_val
+
+    latlong_df.index = initial_df.index  # Need the underlying index to match
+    es.update_dataframe('latlongs', latlong_df)
+    df = to_pandas(es['latlongs'])
+    expected_val = (3, 4)
+    if ks and isinstance(es['latlongs'], ks.DataFrame):
+        expected_val = [3, 4]
+    for col in latlong_df.columns:
+        series = df[col]
+        assert series.iloc[-1] == expected_val
+
+
+def test_update_dataframe_column_order(es):
+    original_column_order = es['customers'].columns.copy()
+
+    df = es['customers'].copy()
+    col = df.pop('cohort')
+    df[col.name] = col
+
+    assert not df.columns.equals(original_column_order)
+    assert set(df.columns) == set(original_column_order)
+
+    es.update_dataframe(dataframe_id='customers', df=df)
+
+    assert es['customers'].columns.equals(original_column_order)
+
+
+def test_update_dataframe_woodwork_initialized(es):
```

Comment:
What exactly are we trying to test here? Is it that Featuretools uses the information in the schema that was passed in the Woodwork dataframe, and does not reinitialize?

If so, I'm thinking maybe we should initialize the df that we pass to update_dataframe with some metadata and confirm that that information is still present when we access the schema through `es['customers'].ww.schema`. 

I might be missing the idea a bit though...

Reply:
Correct, it's testing that passing a Woodwork DataFrame to `update_dataframe` means that you will lose any metadata that was in the original dataframe if it's not in the new one (relevant for secondary time index). 

I did that by changing the metadata in the EntitySet's original DataFrame, but I can also update the metadata in the new dataframe and confirm that it's there as well, since it seems important that any metadata in your new dataframe doesn't get lost too.

Reply:
This does make me wonder if we want to allow other schema change.... like what if we change the index or time index through the new DataFrame? At that point you should probably go through `add_dataframe` instead, right?

Reply:
Ah, I was missing a little detail but I think I'm with you now. I think it might be good to also confirm that any metadata that comes in on the new df gets retained, in addition to verifying that the old metadata is not.

As for changing the index or time index, I was thinking users would call `es['customers'].ww.set_index('new_index')` for that and wouldn't go through `add_dataframe` or `update_data` for that type of change. I guess I would expect that if a user passed in a new dataframe, with WW initialized and the index and/or time index did not match the old values, an error would be raised.

Now that has me questioning if we to allow for different metadata even...

Reply:
Maybe we don't allow a user to pass a Woodwork DataFrame (or if they do we use the current schema and raise a warning telling them that the schema they passed with the new dataframe was ignored) that way we're only allowing updated data that works with the original DataFrame's schema.

Reply:
I'm on the fence here, but leaning toward your last suggestion - maybe for `update_data` we ignore any schema information on the incoming dataframe, and as long as the new data is compatible with the old schema (after handling dtype differences), we are good, otherwise we error.

@rwedge Any thoughts here?

Reply:
My impression of the intent with `update_data` is for the schema to stay the same but the underlying data to change

Reply:
> My impression of the intent with `update_data` is for the schema to stay the same but the underlying data to change

With that in mind, should we just check that the incoming schema is equal to the current schema, and raise an error if not? (for data that has WW initialized)

Reply:
@thehomebrewnerd either that we ignore the incoming Schema and try to initialize the new DataFrame with the current schema. 

So as long as the data a user passes is valid for the current schema it's fine; we don't care about any incoming schema.

Reply:
@tamargrey I guess that's fine too, but I think we should warn the user in that case as you suggested above so they know any new schema information they provided has been lost.

Reply:
pushed the changes with the warning. lmk what you think! Happy to go with the error version if that makes more sense

Reply:
Looks fine to me. 

As I was reviewing, I did notice a couple spelling errors in the docstring for `update_dataframe` that would be good to fix though:
`EntitSet` -> `EntitySet`
`Optionaly` -> `Optionally`

Reply:
Thanks for noticing. Updated the docstring to the following:

> Update the internal dataframe of an EntitySet table, keeping Woodwork typing information the same.
        Optionally makes sure that data is sorted, that reference indexes to other dataframes are consistent, 
        and that last_time_indexes are updated to reflect the new data.